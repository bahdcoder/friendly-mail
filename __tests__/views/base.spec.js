const BaseRenderEngine = require('../../src/Views/Base')

describe('The base render engine', () => {
  it('can be instantiated properly', () => {
    const config = {
      viewEngine: 'handlebars'
    }

    expect(new BaseRenderEngine(config)).toMatchSnapshot()
  })

  it('can get views path', () => {
    const config = {
      viewEngine: 'handlebars'
    }

    const renderEngine = new BaseRenderEngine(config)

    expect(renderEngine._getViewsPath('confirm-email')).toEqual(
      `${process.cwd()}/mails/confirm-email`
    )
  })

  it('can get views path using views config ', () => {
    const config = {
      viewEngine: 'handlebars',
      views: 'server/mails'
    }

    const renderEngine = new BaseRenderEngine(config)

    expect(renderEngine._getViewsPath('confirm-email')).toEqual(
      `${process.cwd()}/server/mails/confirm-email`
    )
  })

  it('can get views content based on viewEngine and views config variables', () => {
    const config = {
      viewEngine: 'handlebars',
      views: '__tests__/__mocks__/mails'
    }

    const edgeConfig = {
      viewEngine: 'edge',
      views: '__tests__/__mocks__/mails'
    }

    expect(
      new BaseRenderEngine(config)._getContent('confirm-email')
    ).toMatchSnapshot()
    expect(
      new BaseRenderEngine(edgeConfig)._getContent('reset-password')
    ).toMatchSnapshot()
  })

  it('can gracefully ignore mail templates that are not found', () => {
    const config = {
      views: '__tests__/__mocks__/mails',
      viewEngine: 'handlebars'
    }

    const base = new BaseRenderEngine(config)
    const content = base._getContent('payment-received')

    expect(content.text).toBeNull()
    expect(content.watchHtml).toBeNull()
    expect(content.html).toMatchSnapshot()
  })
})
