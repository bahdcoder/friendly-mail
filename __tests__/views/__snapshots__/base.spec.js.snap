// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The base render engine can be instantiated properly 1`] = `
BaseRenderEngine {
  "Config": Object {
    "viewEngine": "handlebars",
  },
  "enginesExtensionsMap": Object {
    "edge": "edge",
    "handlebars": "hbs",
  },
}
`;

exports[`The base render engine can get views content based on viewEngine and views config variables 1`] = `
Object {
  "html": "<strong>
  HEY {{ username }}
</strong>
<br>
CONFIRM YOUR EMAIL
",
  "text": "hey {{ username }}, please confirm your email.
",
  "watchHtml": "hey {{ username }}, please confirm your email.
",
}
`;

exports[`The base render engine can get views content based on viewEngine and views config variables 2`] = `
Object {
  "html": "<strong>
  HEY {{ username }}
</strong>
<br>
RESET YOUR PASSWORD
",
  "text": "hey {{ username }}, please RESET YOUR PASSWORD.
",
  "watchHtml": "hey {{ username }}, please RESET YOUR PASSWORD.
",
}
`;

exports[`The base render engine can gracefully ignore mail templates that are not found 1`] = `
"Hello. Your payment of <strong>\${{ price }}</strong> has been received.
"
`;
